src/tests/ftest/launch.py

```diff
@@ -1593,19 +1593,25 @@ def install_debuginfos():
 
     if USE_DEBUGINFO_INSTALL:
         distro_info = detect()
-        yum_args = ["--exclude", "ompi-debuginfo", "openmpi3"]
+        dnf_args = ["--exclude", "ompi-debuginfo", "openmpi3"]
         if "suse" in distro_info.name.lower():
-            yum_args.extend(["libpmemobj1", "python3"])
+            dnf_args.extend(["libpmemobj1", "python3"])
+            # There are no "debug" repos
+            # Packages are found within daos-stack* repos and opensuse* repos
+            cmds.append(
+                ["sudo", "dnf", "debuginfo-install", "--enablerepo='*'", "-y"] +
+                dnf_args + ["daos-server", "gcc"])
         elif "centos" in distro_info.name.lower():
-            yum_args.extend(["libpmemobj", "python36"])
+            dnf_args.extend(["libpmemobj", "python36"])
+            cmds.append(
+                ["sudo", "debuginfo-install", "--enablerepo=*-debuginfo", "-y"] +
```

Comment:
This must be a defect in our test nodes' debuginfo repo installation then as this is not necessary on a stock EL7 system:

```
$ sudo dnf debuginfo-install gcc
enabling base-debuginfo repository
enabling epel-debuginfo repository
…
CentOS-7 - Debuginfo                            0.0  B/s |   0  B     00:00    
CentOS-7 - Debuginfo                            321 kB/s |  86 MB     04:35    
…
Dependencies resolved.
================================================================================
 Package                Arch       Version             Repository          Size
================================================================================
Installing:
 gcc-debuginfo          x86_64     4.8.5-44.el7        base-debuginfo     195 M
Installing dependencies:
 gcc-base-debuginfo     x86_64     4.8.5-44.el7        base-debuginfo     2.9 M

Transaction Summary
================================================================================
Install  2 Packages
```

What does `dnf repolist` report in such a case as yours where it's not finding the repos without some help?

Reply:
→ dnf repolist
...
...
...
repo name                                                                                     status
extras                                                                    CentOS-7 - Extras                                                                                468
repo.dc.hpdd.intel.com_repository_centos-7.9-x86_64-group_                added from: https://repo.dc.hpdd.intel.com/repository/centos-7.9-x86_64-group/                26,041
repo.dc.hpdd.intel.com_repository_daos-stack-el-7-x86_64-stable-local_    added from: https://repo.dc.hpdd.intel.com/repository/daos-stack-el-7-x86_64-stable-local/    19,712

 → dnf repolist --enablerepo="*"
...
...
repo id                                                                   repo name                                                                                     status
base                                                                      CentOS-7 - Base                                                                               10,072
base-debuginfo                                                            CentOS-7 - Debuginfo                                                                           8,361
base-source                                                               CentOS-7 - Base Sources                                                                        2,748
centos-kernel                                                             CentOS LTS Kernels for x86_64                                                                    103
centos-kernel-experimental                                                CentOS Experimental Kernels for x86_64                                                            20
centosplus                                                                CentOS-7 - Plus                                                                                   93
centosplus-source                                                         CentOS-7 - Plus Sources                                                                           13
*epel                                                                     Extra Packages for Enterprise Linux 7 - x86_64                                                13,580
*epel-debuginfo                                                           Extra Packages for Enterprise Linux 7 - x86_64 - Debug                                         2,865
*epel-source                                                              Extra Packages for Enterprise Linux 7 - x86_64 - Source                                        7,195
extras                                                                    CentOS-7 - Extras                                                                                468
extras-source                                                             CentOS-7 - Extras Sources                                                                         76
repo.dc.hpdd.intel.com_repository_centos-7.9-x86_64-group_                added from: https://repo.dc.hpdd.intel.com/repository/centos-7.9-x86_64-group/                26,041
repo.dc.hpdd.intel.com_repository_daos-stack-el-7-x86_64-stable-local_    added from: https://repo.dc.hpdd.intel.com/repository/daos-stack-el-7-x86_64-stable-local/    19,712
updates                                                                   CentOS-7 - Updates                                                                             1,932
updates-source                                                            CentOS-7 - Updates Sources                                                                       197

Reply:
@brianjmurrell, lets leave "--enablerepo=*debuginfo", agree?

Reply:
I think what needs to happen is to file a DCO ticket to have the repos on our test nodes fixed so that this hack isn't necessary and that `dnf debuginfo-install` works the way it does on any pristine EL7 installation without the need for hacks.

Can you write up a DCO ticket describing your need to add the `--enablerepo=*debuginfo` to your `dnf debuginfo-install` command, when it shouldn't be needed.  You can use your `gcc` test case example above in your ticket, and include my example that doesn't need the `--enablerepo=*debuginfo` to show how it is expected to operate.

Reply:
The system I'm using (daos_ci-el7.9)  does not have the modification done by post_provision_config_nodes_EL_7.sh#L50-L58 

The original debuginfo repo file:
```
2021-04-22 22:19:18 ⌚  wolf-193 in /etc/yum.repos.d
→ rpm -qf CentOS-Debuginfo.repo
centos-release-7-9.2009.1.el7.centos.x86_64
```

This means the file CentOS-Debuginfo.repo is just as it was installed
```
 2021-04-22 22:20:10  ⌚ wolf-193 in /etc/yum.repos.d
○ → rpm -V centos-release-7-9.2009.1.el7.centos.x86_64
S.5....T.  c /etc/yum.repos.d/CentOS-Base.repo
S.5....T.  c /etc/yum.repos.d/CentOS-CR.repo
S.5....T.  c /etc/yum.repos.d/CentOS-Media.repo
S.5....T.  c /etc/yum.repos.d/CentOS-Vault.repo
S.5....T.  c /etc/yum.repos.d/CentOS-fasttrack.repo
```
What is inside?

```
2021-04-22 22:20:25 ⌚  wolf-193 in /etc/yum.repos.d                                                                 105/43459]
○ → cat ./CentOS-Debuginfo.repo
# CentOS-Debug.repo
#
# The mirror system uses the connecting IP address of the client and the
# update status of each mirror to pick mirrors that are updated to and
# geographically close to the client.  You should use this for CentOS updates
# unless you are manually picking other mirrors.
#

# All debug packages from all the various CentOS-7 releases
# are merged into a single repo, split by BaseArch
#
# Note: packages in the debuginfo repo are currently not signed
#

[base-debuginfo]
name=CentOS-7 - Debuginfo
baseurl=http://debuginfo.centos.org/7/$basearch/
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Debug-7
enabled=0
```
Which is pretty much the same as the post_provisioning step, the only difference is the name, it preserves the debuginfo suffix. Even if we replace the contents of the file with the post_provisioning, the behaviour persists.

I'd like to avoid going deeper into this, the goal is to avoid ftest to hang when using yum install, adding the flag "--enablerepo=*debuginfo" is a minor inconvenience and could be investigated deeper if is really a priority.

Reply:
@brianjmurrell, thoughts?

Reply:
If you file a DCO ticket documenting the need for the hack, complete with examples of what happens with and without the hack so that somebody reading it can understand the hack and why it's needed, we could go ahead and land this.

In your ticket, make reference to where in the code the hack is being added and that it should be removed when the repo issue is fixed.  Paste the ticket URL to this PR for future cross-reference.

Reply:
https://jira.hpdd.intel.com/browse/DCO-9841